### PR TITLE
docs: describe resource ordering in UI

### LINKED ISSUE #8995: ☂️Ordering dashboard resources
Coder already supports [`order`](https://registry.terraform.io/providers/coder/coder/latest/docs/data-sources/parameter#order) for `coder_paremeter` on the create workspace form. For the sake of completion, we should add ordering to

- [x] #8485

Currently, we are considering [nested `coder_app` buttons](https://github.com/coder/coder/issues/8237) to improve our Dashboard UI.

As the next step, we may improve this further by adding order to the 

- [x] #8487
- [x] #8994
- [x] #7964
- [x] #12074

### PR DESCRIPTION
**Pull request by mtojek**
Fixes: https://github.com/coder/coder/issues/8995

Preview: [link](https://github.com/coder/coder/blob/8995-docs-order/docs/templates/resource-ordering.md)

This PR extends Coder docs with extra content describing resource ordering in the UI. I tried to refer to all the places where the template admin can customize the order of elements/resources.

### CODE REVIEW

File: docs/manifest.json [hunk 1]
```diff
@@ -236,11 +241,6 @@
           "path": "./templates/process-logging.md",
           "state": "enterprise"
         },
-        {
```
Comment (mtojek) [inline docs/manifest.json:239]: Looks like this entry is a duplicate.

**Review by matifali — DISMISSED** [review]
A few suggestion and then good to merge.

File: docs/templates/resource-ordering.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+The resource with the lower `order` is presented before the one with greater
+value. A missing `order` property defaults to 0.
+
+## Using `order` property
+
+### Coder parameters
+
+The `coder_parameter` resource allows specifying the order of parameters in UI
+forms using the `order` property. For example:
```
[the remark this answers is not in the document]
Comment (johnstcn) [reply docs/templates/resource-ordering.md]: Typo: `odrer` should be `order`.

File: docs/templates/resource-ordering.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+```hcl
+resource "coder_parameter" "project_id" {
```
[the remark this answers is not in the document]
Comment (mtojek) [reply docs/templates/resource-ordering.md]: Fixed

File: docs/templates/resource-ordering.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mtojek) [reply docs/templates/resource-ordering.md]: Fixed

File: docs/templates/resource-ordering.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+resource "coder_agent" "secondary" {
+  ...
+
+  order = 2
+}
+```
+
+The agent with the lowest order is presented as the main agent in the workspace
+view.
```
[the remark this answers is not in the document]
Comment (mtojek) [reply docs/templates/resource-ordering.md]: Fixed

File: docs/templates/resource-ordering.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    interval     = 10
+    timeout      = 1
+    order        = 2
+  }
+}
+```
+
+### Applications
+
+Similarly to Cocer agents, `coder_app` resources incorporate the `order`
```
Comment (johnstcn) [inline docs/templates/resource-ordering.md]:
```suggestion
Similarly to Coder agents, `coder_app` resources incorporate the `order`
```

File: docs/templates/resource-ordering.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+property to organize button apps in the app bar within the workspace view.
```
[the remark this answers is not in the document]
Comment (mtojek) [reply docs/templates/resource-ordering.md]: Fixed

File: docs/templates/resource-ordering.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+  option {
+    name        = "US East"
+    description = "Select for east!"
+    value       = "us-east1-a"
+  }
+  ...
+}
+```
+
+### Agent metadata items
```
[the remark this answers is not in the document]
Comment (mtojek) [reply docs/templates/resource-ordering.md]: Fixed

File: docs/templates/resource-ordering.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+## Inherit order from file
+
+### Coder parameter options
+
+The options for Coder parameters maintain the same order as in the file
+structure. This simplifies management and ensures consistency between
+configuration files and UI presentation.
+
+```hcl
+data "coder_parameter" "databse_region" {
```
[the remark this answers is not in the document]
Comment (mtojek) [reply docs/templates/resource-ordering.md]: Fixed

File: docs/templates/resource-ordering.md [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+
+```hcl
+resource "coder_parameter" "project_id" {
+  name         = "project_id"
+  display_name = "Project ID"
+  description  = "Specify cloud provider project ID."
+  order = 2
+}
+
+resource "coder_parameter" "account_id" {
```
[the remark this answers is not in the document]
Comment (mtojek) [reply docs/templates/resource-ordering.md]: Fixed

File: docs/templates/resource-ordering.md (hunk 2 above)
Comment (johnstcn) [inline docs/templates/resource-ordering.md]:
```suggestion
forms using the `order` property. In the below example, `project_id` will appear _before_ `account_id`:
```

File: docs/templates/resource-ordering.md [hunk 10]
```diff
@@ -0,0 +1,181 @@
+# UI Resource Ordering
+
+In Coder templates, managing the order of UI elements is crucial for a seamless
+user experience. This page outlines how resources can be aligned using the
+`order` Terraform property or inherit the natural order from the file.
+
+The resource with the lower `order` is presented before the one with greater
+value. A missing `order` property defaults to 0.
```
Comment (johnstcn) [inline docs/templates/resource-ordering.md:8]:
```suggestion
The resource with the lower `order` is presented before the one with greater
value. A missing `order` property defaults to 0. If two resources have the same
`order` property, the resources will be ordered by property `name`.
```

**Review by johnstcn — APPROVED** [review]

File: docs/templates/resource-ordering.md (hunk 2 above)
Comment (mtojek) [reply docs/templates/resource-ordering.md]: Fixed

File: docs/templates/resource-ordering.md (hunk 10 above)
Comment (mtojek) [reply docs/templates/resource-ordering.md]: Fixed

**Pull request merged**